includes/Core/Util/Activation.php

```diff
@@ -155,9 +155,74 @@ private function get_activation_notice() {
 					ob_start();
 					?>
 					<script type="text/javascript">
-					if( 'undefined' !== typeof sendAnalyticsTrackingEvent ) {
-						sendAnalyticsTrackingEvent( 'plugin_setup', 'plugin_activated' );
-					}
+						if ( 'undefined' !== typeof sendAnalyticsTrackingEvent ) {
+							sendAnalyticsTrackingEvent( 'plugin_setup', 'plugin_activated' );
+						}
+
+						jQuery( document ).ready( function( $ ) {
+							var trackingScriptPresent = !! googlesitekit.admin.trackingOptIn;
+
+							if ( googlesitekit.admin.trackingOptIn ) {
+								$( '#opt-in' ).prop( 'checked', googlesitekit.admin.trackingOptIn );
+							}
+							if ( googlesitekit.admin.proxySetupURL ) {
+								$( '#start-setup-link' ).attr( 'href', googlesitekit.admin.proxySetupURL );
+							}
+
+							$( '#start-setup-link' ).on( 'click' , function() {
+								if ( 'undefined' !== typeof sendAnalyticsTrackingEvent ) {
+									sendAnalyticsTrackingEvent( 'plugin_setup', 'goto_sitekit' );
+								}
+							} );
+
+							$('#opt-in').on( 'change' , function( event ) {
+								if ( event.target.disabled ) {
+									event.preventDefault();
+									return;
+								}
+
+								var checked = event.target.checked;
+
+								var body = {
+									googlesitekit_tracking_optin: checked,
+								};
+								var self = this;
+
+								$( self ).prop( 'disabled', true );
+
+								wp.apiFetch( {
+									path: '/wp/v2/settings',
+									headers: {
+										'Content-Type': 'application/json; charset=UTF-8',
+									},
+									body: JSON.stringify( body ),
+									method: 'POST',
+								} )
+									.then( function() {
+										$(self).prop( 'disabled', null );
+
+										var trackingId = googlesitekit.admin.trackingID;
+										var trackingScriptPresent = $( `script[src="https://www.googletagmanager.com/gtag/js?id=${ trackingId }"]` ).length > 0;
+
+										if ( ! trackingScriptPresent ) {
+											$( 'body' ).append( `
+												\<script async src="https://www.googletagmanager.com/gtag/js?id=${ trackingId }"\>\</script\><?php // phpcs:ignore WordPress.WP.EnqueuedResources.NonEnqueuedScript ?>
+												\<script\>
+													window.dataLayer = window.dataLayer || [];
+													function gtag(){dataLayer.push(arguments);}
+													gtag('js', new Date());
+													gtag('config', '${ trackingId }');
+													window.googlesitekitTrackingEnabled = true;
+												\</script\>
+											` );
+										}
+									} )
+									.catch( function( err ) {
+										$( self ).prop( 'checked', ! checked );
```

Comment:
See above, when this fails, let's set `window.googlesitekitTrackingEnabled` back to the previous value.

Reply:
I've actually set things so that `window.googlesitekitTrackingEnabled = checked;`, only after the request completes. So it should only ever be set according to the checkbox state, _after_ the setting is successfully saved.